### ISSUE TITLE
AUR package

### ISSUE THREAD

**Issue by tuurep** [jannis-baum/vivify#52]
Hi, I've managed to create the AUR package!

https://aur.archlinux.org/packages/vivify

Just now managed to install it using AUR helper [`yay`](https://github.com/Jguer/yay), here's what it looks like when querying with yay:

![image](https://github.com/jannis-baum/vivify/assets/15036238/f0a42dc8-7333-49e3-9824-20ffa538eee9)

Very cool, works perfectly fine :)

You can take a look at the PKGBUILD [here](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=vivify) and if something looks wrong, let me know.

What I can think of is:
1. Project has no license, if you have one in mind it might be wise to add
2. The reason `gcc-libs` is in `depends` is just because it was raised in an error using [`namcap`](https://wiki.archlinux.org/title/namcap) to check the package, so I guess it should be in there and it's good to be safe with the dependencies

Other than that, it's dead simple when I just run your build script :smile:

Maybe we can add link on the README?

**Comment by tuurep** [comment]
On maintenance: for the most part I just have to bump the `pkgver` each time you do a release, I'll be committed to doing that.

**Comment by tuurep** [comment]
Oh, it has a problem. Thought that the AUR package was working because I still had a `vivify-server` process running from a hand-built version, actually the one built with the AUR package hangs on "Launching server ..."

When trying to run the binary `vivify-server`, it outputs

```
Pkg: Error reading from file.
```

Trying to see why this happens.

**Comment by jannis-baum** [comment]
> Oh, it has a problem. Thought that the AUR package was working because I still had a `vivify-server` process running from a hand-built version, actually the one built with the AUR package hangs on "Launching server ..."
> 
> When trying to run the binary `vivify-server`, it outputs
> 
> ```
> Pkg: Error reading from file.
> ```
> 
> Trying to see why this happens.

Hm seems like this is a problem with [pkg](https://www.npmjs.com/package/pkg) but I guess you already know this haha. I probably won't be able to reproduce anything regarding the AUR package on my Mac but in case there is anything I might be able to help with let me know! :)

**Comment by jannis-baum** [comment]
Oh, one more thing: I just noticed that `bash` is listed as a dependency. Why do we need this? The `viv` executable works (at least it should) with any POSIX shell.

And then `yarn` is also listed, is this just for the build process? At runtime it shouldn't need `yarn`. The `build.sh` script also works on any POSIX shell by the way in case that's what bash is listed for.

**Comment by tuurep** [comment]
**tl;dr:** You can make a license release, and I should troubleshoot the `pkg` issue

---

> Should I make a new release to include the license now

Thanks for adding! I think simplest would be to make a release on that so I can update it on the PKGBUILD and we don't have to wait for each others actions and so on :D

> is it so rare to not have gcc-libs that we can leave it out? 

Yeah I have a feeling that it's almost certainly installed, and when I checked what system packages depend on `gcc-libs` it almost looks like it lists my whole OS... I could be wrong, but probably doesn't need to be listed.

> Hm seems like this is a problem with [pkg](https://www.npmjs.com/package/pkg) but I guess you already know this haha.

Yeah actually I just found that out before reading your comment.  Yep this is where the issue lies. So I know that:

```
yarn install
./build.sh
```

makes a working `vivify-server` when done manually, but in the PKGBUILD it makes one that doesn't work, and it has something to do with [`vercel/pkg`](https://github.com/vercel/pkg). I'll try to see if I can fix that.

> `bash` is listed as a dependency

Good catch, I replaced `bash` with `sh` and works all the same. Leaving it out altogether generates a warning on `namcap` so that's why I had bash there, but it should have been sh I believe.

> `yarn` is also listed, is this just for the build process?

Yeah the `yarn` is only a build dependency (`makedepends`, possibly confusing name), I think it's in the right place

**Comment by tuurep** [comment]
> Cool! Is `sh` interpreted as "any POSIX compliant shell", i.e. "whatever you have is good so we don't need to install anything"? That might be good to check because otherwise it could for example be that it unnecessarily ends up installing `dash` or something for someone who already has `zsh`.

I kinda took it for granted that that's what it means but actually verifying that was more confusing than I thought, there's actually no package or group by the name of `sh` in pacman, it's just the symlink `/usr/bin/sh` that points at `bash` for me.

Here's the `namcap` error  on that:

```
vivify E: Dependency bash detected and not included (programs ['sh'] needed in scripts ['usr/bin/viv'])
```

Seems to me that having `sh` in `depends` suppresses that and it's fine, but tbh it's a little unclear to me. Mostly: why is it needed in `depends`? But if that works, I think we can go with that?

Btw the `namcap` output has also this warning (not error) on the `vivify-server` binary:

```
vivify W: ELF file ('usr/bin/vivify-server') lacks PIE.
```

Not sure if that is completely benign/irrelevant or could be a clue :)

**Comment by tuurep** [comment]
Hmmm... found out something that can lead somewhere:

What the PKGBUILD creates is like this:

```
pkg/
  usr/
    bin/
      vivify-server
src/
  bin/
    vivify-server-linux
```

where I intend that `vivify-server-linux` is just copied into `vivify-server` and they're the same file.

But running `vivify-server-linux` inside the `src` dir works, while `vivify-server` doesn't

So it seems like the part where it breaks is here:

```bash
install -Dm755 ./bin/vivify-server-linux "${pkgdir}/usr/bin/vivify-server"

```

**Comment by jannis-baum** [comment]
Nice! So everything works now?🤩

**Comment by tuurep** [comment]
Yes! Can confirm, I pushed it to AUR, installed it and now it works.

You can see the fix commit here:

https://aur.archlinux.org/cgit/aur.git/commit/PKGBUILD?h=vivify&id=101af8e8e1ed81a620e5ecf9da4a56f0b490878c

I don't understand the binary stuff, but something that Arch Build System strips was actually required by `vercel/pkg`?

Also the `namcap` warning is unrelated to this and probably a non-issue.

**Comment by jannis-baum** [comment]
Awesome! So in the README I could write something like

> Install from AUR:
>
> ```sh
> yay -sS vivify
> ```

Is that good?

**Comment by tuurep** [comment]
Drop the lowercase `s`, to install it's: `yay -S vivify`

People use different helpers for AUR, I'm not sure if `yay` is the most popular anymore :smile: I've been seeing that many people prefer one called `paru`

I suggest maybe just putting it like "It's available in the AUR" and a link to [this page](https://aur.archlinux.org/packages/vivify), and people should know how to install it :)
